config/config-api/src/com/thoughtworks/go/config/remote/ConfigRepoConfig.java

```diff
@@ -71,6 +79,16 @@ public String getConfigProviderPluginName() {
         return configProviderPluginName;
     }
 
+    public String getId() {
+        return id;
+    }
+
+    public void setId(String id) {
+        if(StringUtil.isBlank(id))
+            id = null;
```

Comment:
Not sure if the xsd validation takes cate of things like id="                ". There should be a validation on the model for the id then.